adapter/src/ethereum.rs

```diff
@@ -269,6 +278,71 @@ impl Adapter for EthereumAdapter {
         ewt_sign(&wallet, &self.keystore_pwd, &payload)
             .map_err(|err| AdapterError::Adapter(Error::SignMessage(err).into()))
     }
+
+    async fn get_deposit(
+        &self,
+        channel: &ChannelV5,
+        address: &Address,
+    ) -> AdapterResult<Deposit, Self::AdapterError> {
+        let outpace_contract = Contract::from_json(
+            self.web3.eth(),
+            self.config.ethereum_core_address.into(),
+            &OUTPACE_ABI,
+        )
+        .map_err(Error::ContractInitialization)?;
+
+        let erc20_contract =
+            Contract::from_json(self.web3.eth(), channel.token.as_bytes().into(), &ERC20_ABI)
+                .map_err(Error::ContractInitialization)?;
+
+        let total: H256 = tokio_compat_02::FutureExt::compat(async {
+            tokio_compat_02::FutureExt::compat(outpace_contract.query(
+                "deposits",
+                (
+                    H256(*channel.id()).into_token(),
```

Comment:
this is not really readable, could you used the `Token::Address`, for the ChannelId we could leave it as it is. It wont really give us any benefit to use `Token::FixedBytes` (which is the equivalent of bytes32, bytes64, etc. in solidity)